worlds/ff1/Client.py

```diff
@@ -0,0 +1,327 @@
+        if locations_data is None:
+            return
+        locations_checked = []
+        if len(locations_data) > 0xFE and locations_data[0xFE] & 0x02 != 0 and not ctx.finished_game:
+            await ctx.send_msgs([
+                {"cmd": "StatusUpdate",
+                 "status": ClientStatus.CLIENT_GOAL}
+            ])
+            ctx.finished_game = True
+        for location in ctx.missing_locations:
+            # index will be - 0x100 or 0x200
+            index = location
+            if location < 0x200:
+                # Location is a chest
+                index -= 0x100
+                flag = 0x04
+            else:
+                # Location is an NPC
+                index -= 0x200
+                flag = 0x02
+            if locations_data[index] & flag != 0:
+                locations_checked.append(location)
+
+        found_locations = await ctx.check_locations(locations_checked)
+        for location in found_locations:
+            ctx.locations_checked.add(location)
+            location_name = ctx.location_names.lookup_in_game(location)
+            logger.info(
+                f'New Check: {location_name} ({len(ctx.locations_checked)}/'
+                f'{len(ctx.missing_locations) + len(ctx.checked_locations)})')
+
+
+    async def received_items_check(self, ctx: "BizHawkClientContext") -> None:
+        assert self.consumable_stack_amounts, "shouldn't call this function without reading consumable_stack_amounts"
+        write_list: list[tuple[int, list[int], str]] = []
+        items_received_count = await self.read_sram_value_guarded(ctx, items_obtained)
+        if items_received_count is None:
+            return
+        if items_received_count < len(ctx.items_received):
+            current_item = ctx.items_received[items_received_count]
+            current_item_id = current_item.item
+            current_item_name = ctx.item_names.lookup_in_game(current_item_id, ctx.game)
+            if current_item_name in key_items:
+                location = current_item_id - 0xE0
+                write_list.append((location, [1], self.sram))
+            elif current_item_name in movement_items:
+                location = current_item_id - 0x1E0
+                if current_item_name != "Canal":
+                    write_list.append((location, [1], self.sram))
+                else:
+                    write_list.append((location, [0], self.sram))
+            elif current_item_name in no_overworld_items:
+                if current_item_name == "Sigil":
+                    location = 0x28
+                else:
+                    location = 0x12
+                write_list.append((location, [1], self.sram))
+            elif current_item_name in gold_items:
+                gold_amount = int(current_item_name[4:])
+                current_gold = int.from_bytes(await self.read_sram_values_guarded(ctx, gp_location_low, 3), "little")
+                if current_gold is None:
+                    return
```

Comment:
`current_gold` is not something that needs to be checked for `None` - It will never be `None`

What needs to be checked it the result of `await self.read_sram_values_guarded(ctx, gp_location_low, 3)`